examples/Makefile

```diff
@@ -32,6 +31,43 @@ else
     LDD = ldd
 endif
 
+#-------------------------------------------------------------------------------
+# BLAS and LAPACK
+
+ifeq ($(blas),mkl)
+    # MKL on MacOS doesn't include ScaLAPACK; use default.
+    # For others, link with appropriate version of ScaLAPACK and BLACS.
+    ifneq ($(macos),1)
+        ifeq ($(mkl_blacs),openmpi)
+            ifeq ($(blas_int),int64)
+                scalapack_libs ?= -lmkl_scalapack_ilp64 -lmkl_blacs_openmpi_ilp64
+            else
+                scalapack_libs ?= -lmkl_scalapack_lp64 -lmkl_blacs_openmpi_lp64
+            endif
+        else
+            ifeq ($(blas_int),int64)
+                scalapack_libs ?= -lmkl_scalapack_ilp64 -lmkl_blacs_intelmpi_ilp64
+            else
+                scalapack_libs ?= -lmkl_scalapack_lp64 -lmkl_blacs_intelmpi_lp64
+            endif
+        endif
+    endif
```

Comment:
This is ok, but I think we can auto-detect these things so the user doesn't have to set `blas_int` and `mkl_blacs`. That can simplify the main Makefile, too. I'll submit some another change.